adapters/sspBC/sspbc.go

```diff
@@ -224,70 +221,74 @@ func (a *adapter) createBannerAd(bid openrtb2.Bid, ext responseExt, request *ope
 	mcad.SeatBid = make([]openrtb2.SeatBid, 1)
 	mcad.SeatBid[0].Bid = make([]openrtb2.Bid, 1)
 	mcad.SeatBid[0].Bid[0] = bid
+
 	mcMarshalled, err := json.Marshal(mcad)
 	if err != nil {
-		return bid.AdM, err
+		return "", err
 	}
 
-	mcEncoded := base64.URLEncoding.EncodeToString(mcMarshalled)
-
 	bannerData := &templatePayload{
 		SiteId:  ext.SiteId,
 		SlotId:  ext.SlotId,
 		AdLabel: ext.AdLabel,
 		PubId:   ext.PublisherId,
 		Page:    request.Site.Page,
 		Referer: request.Site.Ref,
-		McAd:    mcEncoded,
+		McAd:    base64.URLEncoding.EncodeToString(mcMarshalled),
 	}
 
 	// Prepare banner html, using template file
 	var filledTemplate bytes.Buffer
 	if err := a.bannerTemplate.Execute(&filledTemplate, bannerData); err != nil {
-		return bid.AdM, err
+		return "", err
 	}
 
 	return filledTemplate.String(), nil
 }
 
-func getImpSize(Imp openrtb2.Imp) string {
-	impSize := "1x1"
+func getImpSize(imp openrtb2.Imp) string {
+	if imp.Banner == nil || len(imp.Banner.Format) == 0 {
+		return impFallbackSize
+	}
 
-	if Imp.Banner != nil {
-		areaMax := int64(0)
-		for _, sizeI := range Imp.Banner.Format {
-			areaI := sizeI.W * sizeI.H
-			if areaI > areaMax {
-				areaMax = areaI
-				impSize = fmt.Sprintf("%dx%d", sizeI.W, sizeI.H)
-			}
+	var (
+		areaMax int64
+		impSize = impFallbackSize
+	)
+
+	for _, size := range imp.Banner.Format {
+		area := size.W * size.H
+		if area > areaMax {
+			areaMax = area
+			impSize = fmt.Sprintf("%dx%d", size.W, size.H)
 		}
 	}
 
-	// default fallback
+	if areaMax == 0 {
```

Comment:
Given line 256, I don't think this if statement is necessary? Not that it hurts anything.

Reply:
Thanks. You are right. I wanted to explicitly cover the case when one of the size dimensions is zero. But after time, I fully agree with you it is not necessary.